src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -184,12 +177,6 @@ public bool IsGenerated
 
             var tree = treeFactory.ParseSyntaxTree(filePath, options, text, cancellationToken);
 
-            var root = tree.GetRoot(cancellationToken);
-            if (mode == PreservationMode.PreserveValue && treeFactory.CanCreateRecoverableTree(root))
```

Comment:
because CanCreateRecoverableTree is 'always false', this code cannot execute and what removed.  This allowed CreateRecoverableTree to be removed.